upstream/CVE-2023-3955.json

```diff
@@ -53,12 +53,199 @@
                     ]
                 }
             ]
+        },
+        {
+            "package": {
+                "ecosystem": "eks",
+                "name": "k8s.io/kubelet"
+            },
+            "severity": [
+                {
+                    "type": "CVSS_V3",
+                    "score": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
+                }
+            ],
+            "ranges": [
+                {
+                    "type": "SEMVER",
+                    "events": [
+                        {
+                            "introduced": "1.27.0"
+                        },
+                        {
+                            "fixed": "1.27.4-eks-8ccc7ba"
+                        },
+                        {
+                            "introduced": "1.26.0"
+                        },
+                        {
+                            "fixed": "1.26.7-eks-8ccc7ba"
+                        },
+                        {
+                            "introduced": "1.25.0"
+                        },
+                        {
+                            "fixed": "1.25.12-eks-8ccc7ba"
+                        },
+                        {
+                            "introduced": "1.24.0"
+                        },
+                        {
+                            "fixed": "1.24.16-eks-8ccc7ba"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "package": {
+                "ecosystem": "gke",
+                "name": "k8s.io/kubelet"
+            },
+            "severity": [
+                {
+                    "type": "CVSS_V3",
+                    "score": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
+                }
+            ],
+            "ranges": [
+                {
+                    "type": "SEMVER",
+                    "events": [
+                        {
+                            "introduced": "1.27.0"
+                        },
+                        {
+                            "fixed": "1.27.5-gke.200"
```

Comment:
Did you make sure the kubelet version matches the cluster version?

Reply:
If I understand correctly, 1.24.17-gke.200 is a fixed version for GKE, not kubelet. 

>we recommend that you [manually upgrade](https://cloud.google.com/kubernetes-engine/docs/how-to/upgrading-a-cluster) your cluster and node pools to one of the following GKE versions:
>
>1.24.17-gke.200
>1.25.13-gke.200
>1.26.8-gke.200
>1.27.5-gke.200
>1.28.1-gke.200

I'm wondering if kubelet also has this versioning.

Reply:
in cloud kubernetes services a user cannot bump kubelet by it self. the bump is on `Node poll` where kubelet live.
the vulnerability is in kubelet and the fix is also.

GKE (same as kubernetes) version is the same as kubelet version.

Reply:
>GKE (same as kubernetes) version is the same as kubelet version.

Did you ensure it? I'm just concerned our KBOM detects the kubelet version, but the versioning is different from `1.y.z-gke.xxx`. It results in false detection.

Reply:
yes , all core component versions are the same a kubernetes release version